### PR TITLE
Added OIDC flow to SmartTrader

### PR DESCRIPTION
**Pull request by adrienne-deriv**
Changes:
Added OIDC flow to SmartTrader. This involves:
- A new Callback page at `/en/callback` route which will call `authorize` and set the `client.accounts`
- Deprecating and removing Deriv's legacy iframe `localstorage-sync`
- Adding and checking the cookie `logged_state` to automatically SSO the user when they have previously logged in from other applications

https://github.com/user-attachments/assets/991b0f9c-4db2-4580-b0ec-c320938a56a3

## Type of change

-   [ ] Bug fix
-   [x] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "jsxBracketSameLine": false,
+    "arrowParens": "avoid",
+    "printWidth": 100,
+    "endOfLine": "lf",
+    "object-property-newline": "always",
+    "key-spacing": {
+        "beforeColon": false,
+        "afterColon": true,
+        "mode": "strict"
+    }
```
Comment (adrienne-deriv) [inline .prettierrc:16]: Added prettier config to enable autofix in Vscode

File: src/javascript/_common/auth.js [hunk 2]
```diff
@@ -123,3 +143,47 @@ export const getLogoutHandler = onWSLogoutAndRedirect => {
...
+    if (!isGrowthbookLoaded) {
+        let retryInterval = 0;
+        const interval = setInterval(() => {
+            if (retryInterval > 10) {
+                clearInterval(interval);
+            } else {
+                const isLoaded = Analytics.isGrowthbookLoaded();
+                if (isLoaded) {
+                    _requestSingleSignOn();
+                    clearInterval(interval);
```
Comment (adrienne-deriv) [inline src/javascript/_common/auth.js:177]: This interval is to check if Growthbook has already been loaded. If not, wait for it to be initialised and attempt to SSO if conditions are met

File: src/javascript/app/pages/callback/callback.jsx [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+import { isStorageSupported, removeCookies } from '../../../_common/storage';
+import { urlFor } from '../../../_common/url';
+import { getPropertyValue } from '../../../_common/utility';
+import { getElementById } from '../../../_common/common_functions';
+
+const map_names = {
+    country             : 'residence',
+    landing_company_name: 'landing_company_shortcode',
+};
+
```
Comment (adrienne-deriv) [inline src/javascript/app/pages/callback/callback.jsx:21]: For this component, I just took most of the authentication logic from `logged_in.js` and migrate it to the `Callback` component, which will be called when we receive the tokens from OIDC at `onSignInSuccess`

**Review by hasan-deriv — APPROVED** [review]

**Review by amam-deriv — APPROVED** [review]
[Q] do we have any feature flag to enable/disable OIDC?

**Pull request closed without merging**